### PR TITLE
Trim docs

### PR DESCRIPTION
**Pull request by alsamylkin**
Our documentation is stale, with this change we trim readme to contain only pitch and cleanup comments for godoc.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 - **[Breaking]** `ValueType` and `GetType` functionality is removed in favor of using
   `reflect.Kind`
+- Update documentation: cleanup markup, trim readme and doc.go in
+favor of godoc.
```
Comment (glibsm) [inline CHANGELOG.md]: bad indent, should be two spaces here and also i don't think this is changelog worthy

File: README.md [hunk 2]
```diff
@@ -5,409 +5,11 @@
 [![Coverage Status][cov-img]][cov]
 [![Report Card][report-card-img]][report-card]
```
Comment (glibsm) [inline README.md]: maybe add an emoji prefix to config a-la fx and dig? make a 🔧 or something

File: cache_provider.go [hunk 3]
```diff
@@ -32,10 +32,8 @@ type cachedProvider struct {
 	Provider
 }
 
-// NewCachedProvider returns a provider, that caches values of the underlying Provider p.
-// It also subscribes for changes for all keys that ever retrieved from the provider.
-// If the underlying provider fails to register callback for a particular value, it will
-// return the underlying error wrapped in Value.
+// NewCachedProvider returns a concurrent safe  provider,
```
Comment (glibsm) [inline cache_provider.go]: double space

File: command_line_provider.go [hunk 4]
```diff
@@ -50,8 +52,9 @@ type commandLineProvider struct {
 	Provider
 }
 
-// NewCommandLineProvider returns a Provider that is using command line parameters as config values.
-// In order to address nested elements one can use dots in flag names which are considered separators.
+// NewCommandLineProvider returns a Provider that is using command line
+// parameters as config values. In order to address nested elements one
```
Comment (glibsm) [inline command_line_provider.go]: s/parameters/arguments

File: provider_group.go [hunk 5]
```diff
@@ -34,8 +34,21 @@ func NewProviderGroup(name string, providers ...Provider) Provider {
 	}
 }
 
+// Get iterates through the providers and return the value merged from
+// underlying of providers. The merge strategy for two objects
```
Comment (glibsm) [inline provider_group.go]: `The merge strategy` should start in a new paragraph

File: README.md (hunk 2 above)
Comment (alsamylkin) [inline README.md]: Config is going to fish for values..

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,8 @@
 
 - **[Breaking]** `ValueType` and `GetType` functionality is removed in favor of using
   `reflect.Kind`
+- Update documentation: cleanup markup, trim readme and doc.go in
```
Comment (glibsm) [inline CHANGELOG.md]: are we sure this belongs in changelog?

Comment (breerly) [inline CHANGELOG.md]: Probably not - the changelog should only contain things end users would want to know.

**Pull request merged**